microcosm_fastapi/audit.py

```diff
@@ -0,0 +1,311 @@
+"""
+Audit log support for FastAPI routes.
+
+"""
+from typing import Dict, Any, Optional
+from collections import namedtuple
+from distutils.util import strtobool
+from json import loads
+from json.decoder import JSONDecodeError
+from logging import getLogger
+import json
+from uuid import UUID
+from functools import partial
+from inflection import underscore
+
+from fastapi import Request
+
+from microcosm.api import defaults, typed
+from microcosm.metadata import Metadata
+from microcosm.config.types import boolean
+from microcosm_logging.timing import elapsed_time
+from microcosm_fastapi.utils import AsyncIteratorWrapper
+from microcosm_fastapi.logging_data_map import LoggingInfo
+from microcosm_fastapi.errors import ParsedException
+
+
+DEFAULT_INCLUDE_REQUEST_BODY = 400
+DEFAULT_INCLUDE_RESPONSE_BODY = 400
+ERROR_MESSAGE_LIMIT = 2048
+
+AUDIT_LOGGER_NAME = "audit"
+
+
+AuditOptions = namedtuple("AuditOptions", [
+    "include_request_body",
+    "include_response_body",
+    "include_path",
+    "include_query_string",
+    "log_as_debug",
+])
+
+
+SKIP_LOGGING = "_microcosm_flask_skip_audit_logging"
+
+
+def is_uuid(value):
+    try:
+        UUID(value)
+        return True
+    except Exception:
+        return False
+
+
+def should_skip_logging(request: Request):
```

Comment:
```suggestion
def should_skip_logging(request: Request) -> bool:
```

It'd be nice to give this a once-over and add any easy missing types

Reply:
True - have done as many as I can in this file 👍